test/jdk/valhalla/valuetypes/NullRestrictedTest.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import jdk.internal.vm.annotation.ImplicitlyConstructible;
+import jdk.internal.vm.annotation.NullRestricted;
+
+import org.junit.jupiter.api.Test;
+
+import java.lang.invoke.MethodHandles;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/*
+ * @test
+ * @modules java.base/jdk.internal.vm.annotation
+ *          java.base/jdk.internal.value
+ * @compile -XDenablePrimitiveClasses NullRestrictedTest.java
+ * @run junit/othervm -XX:+EnableValhalla NullRestrictedTest
+ * @run junit/othervm -XX:+EnableValhalla -XX:InlineFieldMaxFlatSize=0 NullRestrictedTest
+ */
+public class NullRestrictedTest {
+    @ImplicitlyConstructible
+    static value class MyValueEmpty {
+    }
+
+    @ImplicitlyConstructible
+    static value class EmptyContainer {
+        @NullRestricted
+        MyValueEmpty empty = new MyValueEmpty();
+    }
+
+    @Test
+    public void lazyInitializedDefaultValue() {
+        // VM lazily sets the null-restricted non-flat field to zero default
+        assertTrue(new EmptyContainer() == EmptyContainer.default);
```

Comment:
The `.default` syntax is translated to `aconst_init` which is going to be removed.
May be use `ValueClass.zeroInstance(EmptyContainer.class)` instead.
Applies to all `.default` usage in this file.